src/pages/productPage/product.js

```diff
@@ -1,10 +1,51 @@
-import { Navbar } from "../../components/navbar/navbar";
+import { Navbar } from "../../components";
+import "./product.css";
+import { Filter, ProductCard } from "./componets";
+import { useFilter } from "../../context/filterContext";
+import axios, { Axios } from "axios";
+import { useState, useEffect } from "react";
+import {
+  sortData,
+  filterByCategory,
+  latestProduct,
+  ratingSlider,
+} from "../../utils";
 
 export function Product() {
+  const { state } = useFilter();
+
+  const [product, setProduct] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get("/api/products");
+        setProduct(response.data.products);
+      } catch (error) {
+        console.log(error.message);
+      }
+    })();
+  }, []);
+
+  const defaultData = [...product];
+  const newSortdata = sortData(defaultData, state.sortBy);
+  const newCategory = filterByCategory(newSortdata, state.category);
+  const filterData = latestProduct(newCategory, state.category);
```

Comment:
Here you can use the concept of composition of functions and use it here
For example: [see this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#function_composition_enabling_piping)

Reply:
thanks for the suggestion ill look into it and try to implement it